src/adapter/graphql/src/scalars/u256.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::LazyLock;
+
+use async_graphql::{InputValueError, InputValueResult, Scalar, ScalarType, Value};
+
+use crate::scalars::BigInt;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+static MAX_U256: LazyLock<num_bigint::BigInt> = LazyLock::new(|| {
+    const ONES_256_BYTES: [u8; 256] = [b'1'; 256];
+    const BINARY_RADIX: u32 = 2;
+
+    let max_u256_as_str = std::str::from_utf8(&ONES_256_BYTES).unwrap();
+
+    // 115792089237316195423570985008687907853269984665640564039457584007913129639935
+    num_bigint::BigInt::parse_bytes(max_u256_as_str.as_bytes(), BINARY_RADIX).unwrap()
+});
+
+#[nutype::nutype(derive(AsRef, Clone, Debug, Into))]
+pub struct U256(num_bigint::BigInt);
```

Comment:
Have you considered using `U256` type from a lib that `alloy` depends on? Since we already link it maybe it would simplify implementation here.

Reply:
Thought about it, but didn't see much advantage -- in the case of the GQL API, we're only reading the value